### ISSUE TITLE
Unhandled Rejection (TypeError): fullscreen error

### ISSUE THREAD

**Issue by wrreee** [snakesilk/react-fullscreen#48]
![image](https://user-images.githubusercontent.com/13641261/101246317-964ddb80-3723-11eb-9c37-4952d02b571f.png)

Hello! is there way to prevent my app from crash in this case? 
I had this error after few fast enter/exit fullscreen mode with hotkey combination.

**Comment by pomle** [comment]
@wrreee, I can't see exactly what happened in your case, but it looks like it might have to do with the initiator of the full-screen toggle coming directly from a user event, like onClick.

For example, when you have the `handle.enter()`, inside `useEffect` it may get toggled for other reasons than the keyboard input for example.

Please share more error details if possible.

**Comment by ChangYinHan** [comment]
hi pomle, I meet the same problem, here is my example.
```
function FScreen(props) {
  const handle = useFullScreenHandle();
  const [flag, setFlag] = useState(0);

   if(props.isopen===true){
    if(flag!==1){
      setFlag(1);
    }
  }

  // I want to set fullscreen on load
  useEffect(()=>{
    if(flag===1){
      handle.enter();
    }
  })
  

  return (
    <div>
      <button onClick={handle.enter}>
        Enter fullscreen
      </button>
      <FullScreen handle={handle} >
        <Router history={browserHistory}>
          <Routes />
        </Router>
      </FullScreen>
    </div>
  )
}

export default FScreen;
```
And I call it by
```
 <PScreen isopen={true}>
    </PScreen>
```
![image](https://user-images.githubusercontent.com/75602404/101612447-9c0a3080-3a45-11eb-939b-5b0f08f7b374.png)

**Comment by pomle** [comment]
@ChangYinHan, you see that is unfortunately not possible to do with the fullscreen API. Browsers do not allow applications to open full screen without it being initiated with a user action as that could be a security issue.

For example, a website would be able to open a webpage that looks like your operating system and trick a user to login to their bank for example.

**Comment by wrreee** [comment]
@pomle yeah u right, but how can i catch this error?

**Comment by pomle** [comment]
> @pomle yeah u right, but how can i catch this error?

@wrreee, I don't know what is best but try wrapping `handle.enter()` in `try/catch`.

**Comment by ChangYinHan** [comment]
> @ChangYinHan, you see that is unfortunately not possible to do with the fullscreen API. Browsers do not allow applications to open full screen without it being initiated with a user action as that could be a security issue.
> 
> For example, a website would be able to open a webpage that looks like your operating system and trick a user to login to their bank for example.

I got it, we should get user permission before change the browser. thanks.

**Comment by wrreee** [comment]
![image](https://user-images.githubusercontent.com/13641261/101619977-0494e580-3a25-11eb-9ae4-d84280a53754.png)
@pomle no(

**Comment by gtnx** [comment]
Yes `try/catch` on the `handle.enter` as the error is 'returned' by the promise `requestFullScreen`. Best optim imho would be a option in `useFullScreenHandle` no?

**Comment by pomle** [comment]
@gtnx what would the option do?

**Comment by gtnx** [comment]
a callback to be called if `requestFullScreen` fails. it could be used for error tracking or displaying an error message to the user

**Comment by pomle** [comment]
@gtnx, when the promise is exposed the user can run `handle.enter().catch()`, so that won't be necessary.

**Comment by gtnx** [comment]
Is the promise exposed?
It does not seem that way:
https://github.com/snakesilk/react-fullscreen/blob/2c0696a58b3368509c2453c3b389f0764360f2ef/src/index.tsx#L28-L36

**Comment by pomle** [comment]
@gtnx that is what will change.

**Comment by pomle** [comment]
This work is not tracked in https://github.com/snakesilk/react-fullscreen/issues/49

I have published pre-version of 0.3.2. It can be installed with `yarn add react-full-screen@0.3.2-0`.

Ping @gtnx, @wrreee, @ChangYinHan

**Comment by pomle** [comment]
This is now available in 1.0.0
